### ISSUE TITLE
Network policy: ingress traffic is allowed when no namespaceSelector/podSelector is specified

### ISSUE THREAD

**Issue by murali-reddy** [weaveworks/weave#3347]
## What you expected to happen?

An user reported this issues in the slack
https://weave-community.slack.com/archives/C2ND76PAA/p1531355773000036

When an ingress network policy applies to a pod, ingress traffic to the pod is blocked by default. Traffic that should be white-listed in specified through `namespaceSelector`, `podSelector` or `ipBlock`. Weave currently does not support `ipBlock` yet, however `namespaceSelector` and `podSelector` specified as network policy peer are honoured.

When a network policy is specified with no `namespaceSelector` or `podSelector` logically we should expect all the ingress traffic to be dropped as no source is specified. However it is observed that traffic is allowed to the pods.

## What happened?

Iptable rule that accepts the ingress traffic should be added only if there are ip's selected by pod/namespace selectors. iptable rule look like below.

```
Chain WEAVE-NPC-INGRESS (1 references)
pkts bytes target     prot opt in     out     source               destination
   1    84 ACCEPT     all  --  *             *0.0.0.0/0            0.0.0.0/0            match-set weave-Q29A#@8a]c!^3}j+e2]M~moXY dst /* anywhere -> pods: namespace: default, selector: app=guestbook */
```
there is no matching ipset for source IP's, thus resulting in allowing all ingress traffic to the pods.

## How to reproduce it?

Apply any network policy with no `namespaceSelector`/`podSelector`. For e.g.

```
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: pod-allow-to-web
  namespace: my-policy
spec:
  podSelector:
    matchLabels:
      app: web
  policyTypes:
  - Ingress
  ingress:
  - from:
    - ipBlock:
        cidr: 100.x.x.x/32
```

**Comment by rade** [comment]
See also #3344 for a similar "fail open" issue.

**Comment by brb** [comment]
Probably this issue can be closed after #3168 has been fixed (I'm planning to do it after the Egress NPC PR has been merged).

**Comment by brb** [comment]
Fixed by #3367
